### PR TITLE
[distributed] rough implementation of `distributed actor` remote resolve factory

### PR DESCRIPTION
**Pull request by kavon**
~Resolves rdar://78484431~

One piece of the puzzle.

### CODE REVIEW

File: stdlib/public/Distributed/DistributedActor.swift [hunk 1]
```diff
@@ -46,9 +46,6 @@ public protocol DistributedActor: AnyActor, Identifiable, Hashable, Codable {
     ///   associated with.
     init(transport: ActorTransport)
 
-    @available(*, deprecated, renamed: "SomeDistributedActor.resolve(_:using:)")
-    init(resolve id: AnyActorIdentity, using transport: ActorTransport) throws
```
Comment (ktoso) [inline stdlib/public/Distributed/DistributedActor.swift:50]: 👍

**Review by ktoso — APPROVED** [review]
Thanks for the cleanup 🙏

File: lib/Sema/TypeCheckDistributed.cpp [hunk 2]
```diff
@@ -127,44 +127,52 @@ bool swift::checkDistributedFunction(FuncDecl *func, bool diagnose) {
   return false;
 }
 
-void swift::checkDistributedActorConstructor(ClassDecl *decl, ConstructorDecl *ctor) {
-  // bail out unless distributed actor, only those have special rules to check here
+/// Ensures that the given constructor will eventually delegate to
+/// self.init(transport:).
+void swift::checkDistributedActorConstructor(ClassDecl *decl,
```
Comment (ktoso) [inline lib/Sema/TypeCheckDistributed.cpp:132]: thx for comment 👍

File: lib/Sema/CodeSynthesisDistributedActor.cpp [hunk 3]
```diff
@@ -342,118 +331,25 @@ createDistributedActor_init_resolve(ClassDecl *classDecl,
...
-    }
-
-    break;
-  }
-  }
-
-  return nullptr;
-}
-
-static void collectNonOveriddenDistributedActorInits(
```
Comment (ktoso) [inline lib/Sema/CodeSynthesisDistributedActor.cpp:385]: thx yeah, thankfully we don't need to worry about any inheritance 🙏

File: lib/Sema/CodeSynthesisDistributedActor.cpp [hunk 4]
```diff
@@ -224,62 +224,71 @@ static void addLocalInit(ClassDecl *classDecl) {
...
-///   switch try transport.resolve(address: address, as: Self.self) {
-///   case .instance(let instance):
-///     self = instance
-///   case .makeProxy:
-///   // TODO: use RebindSelfInConstructorExpr here?
-///     self = <<MAGIC MAKE PROXY>>(address, transport) // TODO: implement this
+/// \verbatim
+/// distributed actor MyActor {
+///   static func resolve(at id: AnyActorIdentity,
+///                       using transport: ActorTransport) throws -> MyActor {
```
Comment (ktoso) [inline lib/Sema/CodeSynthesisDistributedActor.cpp:235]: Minor, I guess I'd go for `resolve(identity, using: transport)` :)

File: lib/Sema/CodeSynthesisDistributedActor.cpp [hunk 5]
```diff
@@ -224,62 +224,71 @@ static void addLocalInit(ClassDecl *classDecl) {
...
+///     TODO: describe body
 ///   }
 /// }
-/// ```
+/// \endverbatim
```
Comment (ktoso) [inline lib/Sema/CodeSynthesisDistributedActor.cpp:239]: thanks for teaching me about \verbatim by the way 👍

**Comment by ktoso** [comment]
Latest resolve synthesis also LGTM -- hopeful that we perhaps could pull it off with a default impl in protocol, but we'll see.

**Comment by ktoso** [comment]
Had a bit of a dupe radar for this, but this also resolves rdar://70840750 👍

**Comment by kavon** [comment]
@swift-ci please smoke test and merge

**Comment by kavon** [comment]
@swift-ci please smoke test and merge

**Comment by kavon** [comment]
The implementation strategy has changed enough to warrant abandoning this PR.

**Pull request closed without merging**